doc/ManifestSpecv0.2.md

```diff
@@ -0,0 +1,315 @@
+# --interactive switch to the installer.
+# Interactive is not supported in this preview (5/24/2020)
+  Interactive: /ShowEula
+
+# Some installers include all localized resources. By specifying a Language switch, winget will pass the value of language to the installer,
+# when the installer is called.
+# Language is not supported in this preview (5/24/2020)
+  Language: en-US
+
+# Installers often write logging files. A user may want to redirect the log to a different location. In order to redirect the log file, the user will
+# pass in a log file path to the installer. For example: --log "%temp%\mylog.txt". The file will be saved as a token in the client: <LOGPATH>. Therefore,
+# if the installer supports log redirection, then the Log switch should be the flag that the installer expects to provide the path to the log
+# file. For example: /LOG=<LOGPATH>.
+# Log must include the <LOGPATH> token.
+  Log: /LOG=<LOGPATH>
+
+# Some installers allow for installing to an alternate location. A user may want to redirect the default install to a different location. 
+# In order to redirect the install location, the user will pass in a installation path to the installer. For example: --installlocation "c:\mytool".
+# The folder path will be saved as a token in the client: <INSTALLPATH>. Therefore, if the installer supports install location redirection,
+# then the InstallLocation switch must be the flag that the installer expects to redirect the installation path.
+# For example: /InstallLocation=<INSTALLPATH>.
+# InstallLocation must include the <INSTALLPATH> token.
+  InstallLocation: /DIR=<INSTALLPATH>
+
+# Some installers allow for additional parameters when doing an update. If winget is doing an update and the
+# UpdateBehavior is specified to be Install, then winget will pass the update switch value on the command line
+# to the installer.
+# Introduced v.02 of manifest spec.
+# Restrictions: [min: 1, max:255]
+  Update: /UsePreviousAppDir
+
+
+# ======================= Installers ==================
+
+# Installers represents the collection of entries that define the actual installer. The installer provides the architecture, url and hash that
+# ensure that the installer has not been tampered with.
+Installers:
+    # Arch is a required field. Arch is the architecture of the tool, and is a required field to ensure that the tool will install correctly.
+    # Supported values: arm, arm64, x86, x64 and neutral
+
+  - Arch: x86
+    # Url is a required field. This provides the path to the installer.
+    # The Url must begin with https:// and followed by a hostname. 
+    # Restrictions: [min: 10, max:2000]
+    Url: https://contosa.net/publiccontainer/contosainstaller.exe
+
+    # Sha256 is a required field. The value is the hash of the installer and used to verify the executable
+    # Restrictions: [valid sha256 hash]
+    Sha256: "69D84CA8899800A5575CE31798293CD4FEBAB1D734A07C2E51E56A28E0DF8C82"
```

Comment:
For this, you'll want to add an entry into `patterns.txt`,

probably something like:
```
(?:Signature|)Sha256: "[0-9A-f]{64}"
```